src/Sulu/Bundle/MediaBundle/Resources/public/js/extensions/sulu-buttons.js

```diff
@@ -0,0 +1,70 @@
+(function() {
+
+    'use strict';
+
+    define([], function() {
+
+        return {
+
+            getButtons: function() {
+                var decoratorDropdown = app.sandbox.sulu.buttons.getApiButton('layout');
+                decoratorDropdown.dropdownItems = {
+                    masonry: {},
+                    table: {}
+                };
+
+                var editCollection = app.sandbox.sulu.buttons.getApiButton('edit');
+                editCollection.title = 'sulu.header.edit-collection';
+                editCollection.disabled = false;
+                editCollection.callback = function() {
+                    app.sandbox.emit('sulu.toolbar.edit-collection');
+                };
+
+                var deleteCollection = app.sandbox.sulu.buttons.getApiButton('delete');
+                deleteCollection.title = 'sulu.header.delete-collection';
+                deleteCollection.callback = function() {
+                    app.sandbox.emit('sulu.toolbar.delete-collection');
+                };
+
+                var moveCollection = app.sandbox.sulu.buttons.getApiButton('move');
+                moveCollection.title = 'sulu.header.move-collection';
+                moveCollection.icon = 'arrows';
+                moveCollection.callback = function() {
+                    app.sandbox.emit('sulu.toolbar.move-collection');
+                };
+
+                return [
+                    {
+                        name: 'mediaDecoratorDropdown',
+                        template: decoratorDropdown
+                    },
+                    {
+                        name: 'editCollection',
+                        template: editCollection
+                    },
+                    {
+                        name: 'deleteCollection',
+                        template: deleteCollection
+                    },
+                    {
+                        name: 'moveCollection',
+                        template: moveCollection
+                    }];
+            },
+
+            getDropdownItems: function() {
+                return [
+                    {
+                        name: 'masonry',
+                        template: {
+                            title: 'sulu.toolbar.masonry',
+                            callback: function() {
+                                app.sandbox.emit('sulu.toolbar.change.masonry');
```

Comment:
@marcelmoosbrugger this is the default event-naming for changing datagrid views (admin bundle defines "sulu.toolbar.change.table" for table view)